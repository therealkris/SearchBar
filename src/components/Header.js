// @flow
import * as React from "react";
import classNames from "classnames";

import styles from "./styles.module.scss";
import SearchBar from "./SearchBar";

export default function Header() {
  return (
    <div className={styles.header}>
      <SearchBar />
      <div className={styles.buttons}>
        <button
            className={classNames(styles.btn, styles.btnSignup)}
            onClick={() => {
              console.log("I'm the sign up button");
            }}
        >
          Sign up
        </button>
        <button
            className={styles.btn}
            onClick={() => {
              console.log("I'm the log in button");
            }}
        >
          Log in
        </button>
        <button
            className={styles.btn}
            onClick={() => {
              console.log("I'm the more button");
            }}
        >
          ...
        </button>
      </div>
    </div>
  );
}
