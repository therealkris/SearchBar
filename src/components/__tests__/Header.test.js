import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";

import Header from "../Header";

test("renders correctly", () => {
  const { container, getByText } = render(<Header />);
  expect(container).toMatchSnapshot();
  expect(getByText(/Sign up/i)).toBeInTheDocument();
});
