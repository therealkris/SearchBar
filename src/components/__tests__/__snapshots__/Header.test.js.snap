// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="search"
    >
      <input
        name="search"
        placeholder="Start typing to search"
        type="search"
      />
    </div>
    <div
      class="buttons"
    >
      <button
        class="btn btnSignup"
      >
        Sign up
      </button>
      <button
        class="btn"
      >
        Log in
      </button>
      <button
        class="btn"
      >
        ...
      </button>
    </div>
  </div>
</div>
`;
